src/plugins/plugin.legend.js

```diff
@@ -377,11 +391,13 @@ export class Legend extends Element {
 		ctx.font = labelFont.string;
 
 		const boxWidth = getBoxWidth(labelOpts, fontSize);
+		const boxHeight = getBoxHeight(labelOpts, fontSize);
+		const height = Math.max(fontSize, boxHeight);
 		const hitboxes = me.legendHitBoxes;
 
 		// current position
 		const drawLegendBox = function(x, y, legendItem) {
-			if (isNaN(boxWidth) || boxWidth <= 0) {
+			if (isNaN(boxWidth) || boxWidth <= 0 || isNaN(boxHeight) || boxHeight < 0) {
```

Comment:
we should make sure getBoxHeight / getBoxWidth return number always, so we don't need to check for isNaN.

Reply:
I updated `getBoxWidth` with similar logic to `getBoxHeight` but I think a NaN width is possible if the `fontSize` gets set to a NaN value